src/features/ui/EtherscanLink/EtherscanLink.tsx

```diff
@@ -17,20 +17,26 @@ export const EtherscanLink: FC<EtherscanLinkProps> = ({
 	address = '',
 	shouldTruncated = true,
 	truncatingStartCharactersCount = 4,
-	className = ''
+	className = '',
+	type = 'address',
+	label
 }) => {
 	const linkPresentation = useMemo(() => {
-		if (!address || !shouldTruncated) return address;
+		if (type === 'address') {
+			if (!address || !shouldTruncated) return address;
 
-		return truncateWalletAddress(address, truncatingStartCharactersCount);
-	}, [shouldTruncated, address, truncatingStartCharactersCount]);
+			return truncateWalletAddress(address, truncatingStartCharactersCount);
```

Comment:
Probably not one for this PR but for reference we could use `truncateAddress` from `zApp-utils`: [truncateAddress](https://github.com/zer0-os/zApp-utils/blob/dcb6e20cc548afbea93501b5f74611fc5103f13b/src/formatting/addresses.ts#L5)

Reply:
Yeah, I agree. I also thought the `getEtherscanUri` itself should also be part of `zApp-utils`